tests/capi/unittest_capi_service_training_offloading.cc

```diff
@@ -0,0 +1,292 @@
+
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+
+#include "ml-api-service-offloading.h"
+
+/**
+ * @brief Internal function to get the config file path.
+ */
+static gchar *
+_get_config_path (const gchar *config_name)
+{
+  const gchar *root_path = g_getenv ("MLAPI_SOURCE_ROOT_PATH");
+
+  /* Supposed to run test in build directory. */
+  if (root_path == NULL)
+    root_path = "..";
+
+  gchar *config_file = g_build_filename (
+      root_path, "tests", "test_models", "config", config_name, NULL);
+
+  return config_file;
+}
+
+/**
+ * @brief Test base class for Database of ML Service API.
+ */
+class MLServiceTrainingOffloading : public ::testing::Test
+{
+  protected:
+  GTestDBus *dbus;
+  GBusType bus_type;
+  int status;
+  ml_service_h client_h;
+  ml_service_h server_h;
+
+  public:
+  /**
+   * @brief Setup method for each test case.
+   */
+  void SetUp () override
+  {
+    g_autofree gchar *services_dir = g_build_filename ("/usr/bin/ml-test/services", NULL);
+
+    dbus = g_test_dbus_new (G_TEST_DBUS_NONE);
+    ASSERT_NE (nullptr, dbus);
+
+    g_test_dbus_add_service_dir (dbus, services_dir);
+
+    g_test_dbus_up (dbus);
+#if defined(ENABLE_GCOV)
+    bus_type = G_BUS_TYPE_SYSTEM;
+#else
+    bus_type = G_BUS_TYPE_SESSION;
+#endif
+
+    /* should start receiver */
+    g_autofree gchar *receiver_config
+        = _get_config_path ("training_offloading_receiver.conf");
+    status = ml_service_new (receiver_config, &client_h);
+    ASSERT_EQ (status, ML_ERROR_NONE);
+
+    g_autofree gchar *sender_config = _get_config_path ("training_offloading_sender.conf");
+    status = ml_service_new (sender_config, &server_h);
+    ASSERT_EQ (status, ML_ERROR_NONE);
+  }
+
+  /**
+   * @brief Teardown method for each test case.
+   */
+  void TearDown () override
+  {
+    if (dbus) {
+      g_test_dbus_down (dbus);
+      g_object_unref (dbus);
+    }
+
+    status = ml_service_destroy (client_h);
+    EXPECT_EQ (ML_ERROR_NONE, status);
+
+    status = ml_service_destroy (server_h);
+    EXPECT_EQ (ML_ERROR_NONE, status);
+  }
+
+  /**
+   * @brief Get available port number.
+   */
+  static guint _get_available_port (void)
+  {
+    struct sockaddr_in sin;
+    guint port = 0;
+    gint sock;
+    socklen_t len = sizeof (struct sockaddr);
+
+    sin.sin_family = AF_INET;
+    sin.sin_addr.s_addr = INADDR_ANY;
+    sin.sin_port = htons (0);
+
+    sock = socket (AF_INET, SOCK_STREAM, 0);
+    EXPECT_TRUE (sock > 0);
+    if (sock < 0)
+      return 0;
+
+    if (bind (sock, (struct sockaddr *) &sin, sizeof (struct sockaddr)) == 0) {
+      if (getsockname (sock, (struct sockaddr *) &sin, &len) == 0) {
+        port = ntohs (sin.sin_port);
+      }
+    }
+    close (sock);
+
+    EXPECT_TRUE (port > 0);
+    return port;
+  }
+};
+
+
+/**
+ * @brief Callback function for reply test.
+ */
+static void
+_receive_trained_model_cb (ml_service_event_e event, ml_information_h event_data, void *user_data)
+{
+  switch ((int) event) {
```

Comment:
unnecessary casting (ml_service_event_e is enumeration)
`switch (event) {`

Reply:
okay.